src/Features/Core/Portable/DocumentHighlighting/AbstractDocumentHighlightsService.cs

```diff
@@ -30,20 +30,18 @@ internal abstract partial class AbstractDocumentHighlightsService : IDocumentHig
             var client = await RemoteHostClient.TryGetClientAsync(document.Project, cancellationToken).ConfigureAwait(false);
             if (client != null)
             {
-                var result = await client.RunRemoteAsync<IList<SerializableDocumentHighlights>>(
-                    WellKnownServiceHubService.CodeAnalysis,
-                    nameof(IRemoteDocumentHighlights.GetDocumentHighlightsAsync),
+                var result = await client.TryInvokeAsync<IRemoteDocumentHighlightsService, ImmutableArray<SerializableDocumentHighlights>>(
                     solution,
-                    new object[]
-                    {
-                        document.Id,
-                        position,
-                        documentsToSearch.Select(d => d.Id).ToArray()
-                    },
+                    (service, solutionInfo, cancellationToken) => service.GetDocumentHighlightsAsync(solutionInfo, document.Id, position, documentsToSearch.SelectAsArray(d => d.Id), cancellationToken),
                     callbackTarget: null,
                     cancellationToken).ConfigureAwait(false);
 
-                return result.SelectAsArray(h => h.Rehydrate(solution));
+                if (!result.HasValue)
```

Comment:
for my understanding. this indicates some issue that we will have already reported through some other means (i.e. yellowbar?)

Reply:
Yes, the issue is communicated via info-bar to the user, so the feature does not need to report it again. It just needs to recover from the failure in a reasonable way.